seqlike/Mutation.py

```diff
@@ -0,0 +1,136 @@
+        mutant_letter: Optional[str] = None,
+    ):
+        """Initialize a Mutation object.
+
+        We initialize a mutation object using the following order of priority:
+
+        1. If a mutation_string in the form of "K15R" (wt-position-mutation)
+           or "15-" (position-mutation)
+           or "^15R" is provided,
+           we use the mutation_string itself to initialize the object.
+        2. Otherwise, both a position and a mutant letter must be provided.
+        """
+        if position is None and mutant_letter is None and mutation_string is None:
+            raise ValueError("At least (position, mutant_letter) or (mutation_string) must be provided!")
+        if mutation_string:
+            wt_letter, position, mutant_letter = parse(mutation_string)
+
+        self.wt_letter = wt_letter
+        if position is None:
+            raise ValueError("Mutations must have a position specified!")
+        self.position = position
+
+        if mutant_letter is None:
+            raise ValueError("Mutations must have a mutant letter specified!")
+        self.mutant_letter = mutant_letter
+        self._one_indexed = False  # internal flag, not to be modified.
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __str__(self):
+        wt_letter = self.wt_letter
+        if self.wt_letter is None:
+            wt_letter = ""
+        return f"{wt_letter}{self.position}{self.mutant_letter}"
+
+    def __add__(self, other: int):
+        position = self.position + other
+        obj = deepcopy(self)
+        obj.position = position
+        return obj
+
+    def __sub__(self, other: int):
+        return self.__add__(other * -1)
+
+    def __gt__(self, other: "Mutation"):
+        if self.position != other.position:
+            return self.position > other.position
+        else:
+            return self.mutant_letter > other.mutant_letter
+
+    def __lt__(self, other: "Mutation"):
+        if self.position != other.position:
+            return self.position < other.position
+        else:
+            return self.mutant_letter < other.mutant_letter
+
+    def __eq__(self, other: "Mutation"):
+        return (
+            self.position == other.position
+            and self.mutant_letter == other.mutant_letter
+            and self.wt_letter == other.wt_letter
+        )
+
+
+class Substitution(Mutation):
+    pass
+
+
+class Deletion(Mutation):
+    def __init__(self, mutation_string=None, position=None, mutant_letter=None):
+        super().__init__(mutation_string=mutation_string, position=position, mutant_letter="-")
+        # Simply override mutant_letter regardless of what users pass in.
+        self.mutant_letter = "-"
+
+
+class Insertion(Mutation):
+    def __str__(self):
+        return f"^{self.position}{self.mutant_letter}"
+
+
+from seqlike.alphabets import STANDARD_AA_SET, STANDARD_NT_SET
+
+STANDARD_LETTERS = STANDARD_AA_SET.union(STANDARD_NT_SET)
+
+
+def parse(mutation_string: str):
```

Comment:
I think this should be a method of Mutation- as far as I can tell, it's only called in its constructor.  In the spirit of "There should be one– and preferably only one –obvious way to do it", I think it makes sense to make the obvious thing instantiating a Mutation object.

Reply:
I thought about this comment, and tried to retrace my thought process in keeping this function separate. The original idea, I think, was to keep the `Mutation.__init__()` clean, hence the contents of the function were kept separate from `__init__()`. I also desired to be able to call on the function in a standalone fashion if possible. Not sure why, but it felt like a sane decision to take. I guess we can make it a class method, but I think that would make calling on `parse` standalone a bit more difficult?

Reply:
Stand-alone or not is a matter of taste I suppose.  I'd rather make Mutations and use attributes to get what I need, but I since this pattern is currently supported, I'm ok leaving it as a standalone function.  I agree that it could be useful.

However, I'd **strongly** recommend changing the name to something more specific than "parse".  How many parse functions are there in the world? =)